pkg/vault/util.go

```diff
@@ -0,0 +1,79 @@
+package vault
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+// Login TODO
+func Login(vaultClient VaultType, vaultConfig *Config) error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	avpConfigPath := filepath.Join(home, ".avp", "config.json")
+	if _, err := os.Stat(avpConfigPath); err == nil {
+		// Open our jsonFile
+		jsonFile, err := os.Open(avpConfigPath)
+		if err != nil {
+			fmt.Println(err)
+		}
+		// defer the closing of our jsonFile so that we can parse it later on
+		defer jsonFile.Close()
+
+		byteValue, _ := ioutil.ReadAll(jsonFile)
+
+		var result map[string]interface{}
+		json.Unmarshal([]byte(byteValue), &result)
+
+		vaultConfig.VaultAPIClient.SetToken(result["vault_token"].(string))
+		tokenself, _ := vaultConfig.VaultAPIClient.Auth().Token().LookupSelf()
+		expireTime := tokenself.Data["expire_time"]
+		today := time.Now()
+		myDate, _ := time.Parse("2021-01-08T08:21:27.195176037Z", expireTime.(string))
```

Comment:
What is the specific date string used for here?

Reply:
`Parse parses a formatted string and returns the time value it represents. The layout defines the format by showing how the reference time, defined to be` - https://golang.org/pkg/time/#Parse